run.py

```diff
@@ -56,36 +115,119 @@ def tweet_media_urls(tweet_status):
+
+        return get_media_jpg_or_gif(media)
     else:
-        return []
+        return {}
+
+def get_media_jpg_or_gif(media):
 
+    a=[ { 'filename': f"{item['id_str']}.jpg", 
+          'url': f"{item['media_url']}?format=jpg&name=large" }
+        for item in media if item['type'] == 'photo' ]
+
+    b=[ { 'filename': f"{item['id_str']}.mp4", 
+          'url': f"{item['video_info']['variants'][0]['url']}" }
+        for item in media
+        if item['type'] == 'video' or item['type'] == 'animated_gif' ]
+
+    for item in media:
+        if item['type'] == 'photo': continue
+        if item['type'] == 'animated_gif': continue
+        if item['type'] == 'video': continue
+        from pprint import pprint as pp
+        pp("Unhandled media type")
+        pp(item["type"])
+#            import code
+#            code.interact(local=dict(globals(), **locals()))
+
+    return a+b
 
 def create_folder(output_folder):
+    "Create a folder if it doesn't exist. Return modification time of .timestamp if it does."
     if not os.path.exists(output_folder):
         os.makedirs(output_folder)
-
+        timestamp = 0
+    else:
+        try:
+            timestamp = os.stat(os.path.join(output_folder, '.timestamp')).st_mtime
+        except:
+            timestamp = 0
+    return timestamp
 
 def download_images(status, num_tweets, output_folder):
-    create_folder(output_folder)
+    timestamp = create_folder(output_folder)
+    ts_file = os.path.join(output_folder, '.timestamp')
     downloaded = 0
 
     for tweet_status in status:
+
+        # XXX The tweet time is ahead by one hour. What the heck!?
+        # Daylight savings? If UTC, should be 7 or 8 hours offset. 
+        # Tried UTC timezone but it didn't help.
+#        tweet_status.created_at = \
+#            tweet_status.created_at.astimezone(timezone.utc)
+
+#        from pprint import pprint as pp
+#        pp(tweet_status._json)
+#        import code
+#        code.interact(local=dict(globals(), **locals()))
+        
         if downloaded >= num_tweets:
+            print('Stopping after downloading ' + str(num_tweets) + ' tweets.' )
             break
 
-        for count, media_url in enumerate(tweet_media_urls(tweet_status)):
-            # Only download if there is not a picture with the same name in the folder already
-            created = tweet_status.created_at.strftime('%d-%m-%y at %H.%M.%S')
-            file_name = "{}_({}).jpg".format(created, count + 1)
-            if not os.path.exists(os.path.join(output_folder, file_name)):
+
+        # Metadata for creating the filename, image comment, timestamp
+        created = tweet_status.created_at.strftime('%y-%m-%d at %H.%M.%S %Z')
+        tweet_id = tweet_status.id_str
+        full_text = tweet_status.full_text
+
+        # Creation time of tweet as seconds.nanoseconds since The Epoch.
+        ctime = tweet_status.created_at.strftime('%s') 
+        ctime = float(ctime)
+        print(ctime)
```

Comment:
I think it would be better to not print the ctime every time. It is too much noise in the output.

Reply:
Will remove. That was mainly for debugging. I had some offset error happening (timezone or daylight savings?). It is slightly off, which could theoretically affect the .timestamp file.  (While this could be debugged and fixed, it seems to makes more sense to research and implement a Twitter "Cursor".)